jvm-packages/python/sparkxgb/xgboost.py

```diff
@@ -0,0 +1,210 @@
+#
+# Copyright (c) 2018 by Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from pyspark import SparkContext, keyword_only
+from pyspark.ml.util import JavaMLWritable, JavaPredictionModel
+from pyspark.ml.wrapper import JavaEstimator, JavaModel
+from sparkxgb.util import XGBoostReadable
+
+
+class JavaParamsOverrides(object):
+    """
+    Mixin for overriding methods derived from JavaParams.
+    """
+
+    # Override the "_from_java" method, so we can read our objects.
+    @classmethod
+    def _from_java(cls, java_stage):
+        """
+        Given a Java object, create and return a Python wrapper of it.
+        Used for ML persistence.
+        """
+
+        # Create a new instance of this stage.
+        py_stage = cls()
+
+        # Load information from java_stage to the instance.
+        py_stage._java_obj = java_stage
+        py_stage._create_params_from_java()
+        py_stage._resetUid(java_stage.uid())
+        py_stage._transfer_params_from_java()
+
+        return py_stage
+
+
+class XGBoostClassifier(JavaParamsOverrides, JavaEstimator, JavaMLWritable, XGBoostReadable):
+    """
+    A PySpark implementation of ml.dmlc.xgboost4j.scala.spark.XGBoostClassifier.
+    """
+
+    @keyword_only
+    def __init__(self, alpha=0.0, baseMarginCol=None, baseScore=0.5, checkpointInterval=-1, checkpointPath='',
```

Comment:
Does it support the addition of booster params other than the general params like interaction constraints and monotone_constraints?

The 0.8 snapshot of xgboost supports it. I am planning to use this wrapper for my project, if it supports addition of other params.

Reply:
Will this permit the extraction of information gain for each feature the model ends up using?

Reply:
Will this permit the extraction of information gain for each feature the model ends up using?